### PR TITLE
convert  distribution.js class comp to a functional component #3394

### PR DESCRIPTION
**Pull request by aravind1078**
#### What does this PR do?
convert `src/js/components/Distribution/Distribution.js` class component into a functional component.
#### What are the relevant issues?
#3394 
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/Distribution/Distribution.js [hunk 1]
```diff
@@ -15,107 +15,104 @@ Value.propTypes = {
...
+          {children}
+        </Distribution>
+      </Box>
+    );
   }
-}
+  return null;
+};
+
+Distribution.defaultProps = {
```
Comment (ShimiSun) [inline src/js/components/Distribution/Distribution.js]: Normally I'll suggest assigning these values as part of the props deconstruction and avoid the usage of `defaultProps`, but the content looks a little too busy to be assigned there (lines 19-26). I'm approving this PR, but I wanted to see if anyone else has other thoughts here.

Comment (ShimiSun) [reply src/js/components/Distribution/Distribution.js]: I'm answering myself here, but I do think this should be moved up to the component declaration and used while deconstructing the props. thanks!

Comment (aravind1078) [reply src/js/components/Distribution/Distribution.js]: @ShimiSun  did the required changes, But not sure why readme.tests are failing?

File: src/js/components/__tests__/__snapshots__/README-test.js.snap [hunk 2]
```diff
@@ -4740,7 +4735,7 @@ string
 Required. Array of objects containing a value. The array should already be
       sorted from largest to smallest value. The caller can put other
       properties in the object. The children function will be called to
-      render the contents of each value. Defaults to \`[]\`.
+      render the contents of each value.
```
Comment (ShimiSun) [inline src/js/components/__tests__/__snapshots__/README-test.js.snap]: where did this change come from? can you clean up your branch and remove this snapshot, unless you think it should be here.

Comment (aravind1078) [reply src/js/components/__tests__/__snapshots__/README-test.js.snap]: @ShimiSun if possible can you please review this ?

**Comment by ShimiSun** [comment]
Thank you for this contribution! I'll be closing this PR as the conversion for hooks on the Distribution component was made on another PR. but we appreciate your efforts here.

**Pull request closed without merging**